linera-version/src/version_info.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    borrow::Cow,
+    io::{Read as _, Write as _},
+    path::PathBuf,
+};
+
+#[derive(
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    async_graphql::SimpleObject,
+    serde::Deserialize,
+    serde::Serialize,
```

Comment:
It would be nice to not include this in `build.rs`.

nit: Also, can we use imports so that it fits on one line? (As was done in 7b9aa0152d and likely lost in the rebase)

Reply:
We can, but the imports would add another several lines — and for the other part of this comment I'd have to `cfg` them all separately (or pull in `cfg_if`).  I think this is the lesser of the two evils, my standing readability concerns around using `SimpleObject` and similarly vaguely-named imports unqualified aside.